src/components/GameComponent.jsx

```diff
@@ -0,0 +1,148 @@
+import { useState, useEffect, useCallback } from "react"
+import { useGameState } from "../zustand/useGameState"
+import { CompassAnimation } from "./CompassAnimation"
+import { useAudio } from "./UseAudio"
+import "/src/css/gameComponent.css"
+
+export const GameComponent = ({ username }) => {
+  const { gameState, setGameState } = useGameState()
+  const [showInfo, setShowInfo] = useState({})
+  const [showOptions, setShowOptions] = useState(true)
+  const [isLoading, setIsLoading] = useState(true)
+  const { isPlaying, toggleAudio } = useAudio()
+  const [history, setHistory] = useState([])
+  const [showHistory, setShowHistory] = useState(false)
+
+  const startGame = useCallback(() => {
+    setIsLoading(true)
+    fetch("https://labyrinth.technigo.io/start", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        username: username,
+      }),
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        setGameState(data)
+        setShowInfo({})
+        setShowOptions(new Array(data.actions.length).fill(true))
+      })
+      .catch((error) => {
+        console.error("Error starting game:", error)
+      })
+      .finally(() => {
+        setTimeout(() => {
+          setIsLoading(false)
+        }, 2000)
+      })
+  }, [setGameState, username])
```

Comment:
One of the requirements this week is "Perform asynchronous operations in the app store" so this could be broken out to be part of the Zustand store actions

Reply:
it should be fixed! :)